pkg/client/k8s.go

```diff
@@ -621,3 +624,71 @@ func (k K8sClient) GetJVs(volNames []string, rType util.ReturnType, labelSelecto
 	}
 	return nil, nil, errors.New("invalid return type")
 }
+
+// GetZFSVols returns a list or a map of ZFSVolume depending upon rType & options
+func (k K8sClient) GetZFSVols(volNames []string, rType util.ReturnType, labelSelector string, options util.MapOptions) (*zfs.ZFSVolumeList, map[string]zfs.ZFSVolume, error) {
+	config := os.Getenv("KUBECONFIG")
```

Comment:
(for much later) (nitpick) Can initializing config be avoided here & `K8sClient.K8sCS` be used instead, via some utility methods like `BuildFromConfig`, etc, as this method is taking `k` as implicit argument but it's not using it.

I understand that the other option for this is going back to RESTclient, is a worse approach from many angles, we'd need to brainstorm after the release. I believe the current approach(your implementation) is the most stable & simple way forward for now.